lm_eval/models/gpt2.py

```diff
@@ -10,36 +10,21 @@ def __init__(self, device="cpu"):
         self.device = torch.device(device)
         self.gpt2 = transformers.GPT2LMHeadModel.from_pretrained('gpt2').to(self.device)
         self.gpt2.eval()
-        self.tokenizer = transformers.GPT2Tokenizer.from_pretrained('gpt2')
+        self.tokenizer = transformers.GPT2TokenizerFast.from_pretrained('gpt2')
 
     @classmethod
     def create_from_arg_string(cls, arg_string):
         args = utils.simple_parse_args_string(arg_string)
         return cls(device=args.get("device", "cpu"))
 
-    def generate(self, context, max_gen_length, truncate=True):
-        # when too long to fit in context, truncate from the left
-        context_tensor = torch.tensor([self.tokenizer.encode(context.strip())[max_gen_length - 1024:]], dtype=torch.long).to(self.device)
-        res = self.gpt2.generate(
-            context_tensor,
-            # TODO: change to have until rather than using eos_token_id
-            eos_token_id=self.tokenizer.eos_token_id,
-            do_sample=False,
-            max_length=self.num_tokens(context) + max_gen_length,
-        )
-
-        # chop off the prompt and the final eos token
-        return self.tokenizer.decode(res[0][min(1024 - max_gen_length, len(context_tensor[0])):-1]).strip()
-
     def loglikelihood(self, context, continuation, truncate=True):
```

Comment:
Clarify whether context+continuation would have some space / word delimiter in between. Alternatively, add_prefix_space=True for encoding continuation.

Reply:
Otherwise we might be inputs like "The object is anapple".

Reply:
The doc string says `If there is a word boundary, the space should be in the continuation. For example, context="hello" continuation=" world" is correct.` Can you clarify how this fails to address your concern?

Reply:
Ah you're right, I missed that.